### PR TITLE
Refactor strings and ints into enum.Enum

### LINKED ISSUE #644: [refactor] `enum.Enum` vs. strings and ints
*I didn't follow the issue templates because this is a refactor, there is no bug or new feature associated.* 

There are many string constants and ints out there, and enums started to be used (for instance in #572).

I think it would be nice to clean up those constants with enums to make the code clearer.

---

Example: the labels normal/anomalous are ints 0/1 respec. 

One will find raw 0/1 in the depths of the code, which is fine when you know what it means, but some authors will define normal/anomalous as 1/0 actually, which creates confusion and you may take a while to realize it.

### PR DESCRIPTION
**Pull request by WenjingKangIntel**
# Description

- Provide a summary of the modification as well as the issue that has been resolved. List any dependencies that this modification necessitates.
Code refactor: add the following types to replace original string constants.

  - LabelType,
  - DirType,
  - SupportedDataType
  - AnomalyMapGenerationMode
  - UpscaleImageMode
  - VisualizeMode

- Fixes #644

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Refactor (non-breaking change which refactors the code base)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://github.com/openvinotoolkit/anomalib/blob/main/CONTRIBUTING.md) of this project.
- [ ] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] I have added a summary of my changes to the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) (not for minor changes, docs and tests).

### CODE REVIEW

**Review by ashwinvaidya17 — CHANGES_REQUESTED** [review]
Thanks for your efforts! I have a few comments.

File: src/anomalib/pre_processing/tiler.py [hunk 1]
```diff
@@ -170,7 +178,7 @@ def __init__(
                 "Please ensure stride size is less than or equal than tiling size."
             )
 
-        if self.mode not in ("padding", "interpolation"):
+        if self.mode not in (UpscaleImageMode.PADDING, UpscaleImageMode.INTERPOLATION):
```
[the remark this answers is not in the document]
Comment (WenjingKangIntel) [reply src/anomalib/pre_processing/tiler.py:181]: Thanks for the comments. Fixed.

File: src/anomalib/pre_processing/tiler.py [hunk 2]
```diff
@@ -111,7 +119,7 @@ def downscale_image(image: Tensor, size: tuple, mode: str = "padding") -> Tensor
         Tensor: Downscaled image
     """
     input_h, input_w = size
-    if mode == "padding":
+    if mode == UpscaleImageMode.PADDING:
```
[the remark this answers is not in the document]
Comment (WenjingKangIntel) [reply src/anomalib/pre_processing/tiler.py:122]: Thanks for the comments. Fixed.

File: src/anomalib/pre_processing/tiler.py [hunk 3]
```diff
@@ -79,12 +87,12 @@ def upscale_image(image: Tensor, size: tuple, mode: str = "padding") -> Tensor:
     image_h, image_w = image.shape[2:]
     resize_h, resize_w = size
 
-    if mode == "padding":
+    if mode == UpscaleImageMode.PADDING:
```
[the remark this answers is not in the document]
Comment (WenjingKangIntel) [reply src/anomalib/pre_processing/tiler.py:90]: Thanks for the comments. Fixed.

File: src/anomalib/post_processing/visualizer.py [hunk 4]
```diff
@@ -122,9 +130,9 @@ def visualize_image(self, image_result: ImageResult) -> np.ndarray:
         Returns:
             The full or simple visualization for the image, depending on the specified mode.
         """
-        if self.mode == "full":
+        if self.mode == VisualizeMode.FULL:
```
[the remark this answers is not in the document]
Comment (WenjingKangIntel) [reply src/anomalib/post_processing/visualizer.py:133]: Thanks for the comments. Fixed.

File: src/anomalib/models/reverse_distillation/anomaly_map.py [hunk 5]
```diff
@@ -37,7 +46,7 @@ def __init__(self, image_size: ListConfig | tuple, sigma: int = 4, mode: str = "
         self.sigma = sigma
         self.kernel_size = 2 * int(4.0 * sigma + 0.5) + 1
 
-        if mode not in ("add", "multiply"):
+        if mode not in (AnomalyMapGenerationMode.ADD, AnomalyMapGenerationMode.MULTIPLY):
```
[the remark this answers is not in the document]
Comment (WenjingKangIntel) [reply src/anomalib/models/reverse_distillation/anomaly_map.py:55]: Thanks for the comments. Fixed.

File: src/anomalib/data/utils/image.py [hunk 6]
```diff
@@ -17,6 +18,13 @@
 from torchvision.datasets.folder import IMG_EXTENSIONS
 
 
+class LabelType(int, Enum):
```
[the remark this answers is not in the document]
Comment (WenjingKangIntel) [reply src/anomalib/data/utils/image.py]: Thanks for the comments. Fixed.

File: src/anomalib/data/__init__.py [hunk 7]
```diff
@@ -25,6 +26,20 @@
 logger = logging.getLogger(__name__)
 
 
+class SupportedDataType(str, Enum):
```
Comment (samet-akcay) [inline src/anomalib/data/__init__.py:29]:
```suggestion
class DataFormat(str, Enum):
```

Comment (samet-akcay) [reply src/anomalib/data/__init__.py:29]: the rest of anomalib refers to this as `format`, so it would be more consistent if this object is named as `DataFormat`

File: src/anomalib/data/utils/label.py [hunk 8]
```diff
@@ -0,0 +1,8 @@
+from enum import Enum
+
+
+class LabelType(int, Enum):
```
Comment (samet-akcay) [inline src/anomalib/data/utils/label.py:4]: How about `LabelName`? 

```suggestion
class LabelName(int, Enum):
```

File: src/anomalib/post_processing/visualizer.py [hunk 9]
```diff
@@ -61,16 +62,23 @@ def __post_init__(self) -> None:
             self.anomalous_boxes = self.pred_boxes[self.box_labels.astype(bool)]
 
 
+class VisualizeMode(str, Enum):
```
Comment (samet-akcay) [inline src/anomalib/post_processing/visualizer.py:65]:
```suggestion
class VisualizationMode(str, Enum):
```

File: src/anomalib/pre_processing/tiler.py [hunk 10]
```diff
@@ -15,6 +16,13 @@
 from torch.nn import functional as F
 
 
+class UpscaleImageMode(str, Enum):
```
Comment (samet-akcay) [inline src/anomalib/pre_processing/tiler.py:19]:
```suggestion
class ImageUpscaleMode(str, Enum):
```

**Review by samet-akcay — CHANGES_REQUESTED** [review]
Thanks for the PR @WenjingKangIntel. I've got few comments.

File: src/anomalib/data/__init__.py (hunk 7 above)
Comment (WenjingKangIntel) [reply src/anomalib/data/__init__.py]: all fixed.

**Review by samet-akcay — COMMENTED** [review]
Thanks @WenjingKangIntel!

I've just noticed one discrepancy between the new Enums as shown below:

`LabelName.ANOMALOUS` vs `DirType.ABNORMAL`. It would be good to have consistent naming for these Enums. I also checked the library where we use `anomalous` and `abnormal` interchangibly. We will probably need to change that in the future.

It might be an idea to change `LabelName.ANOMALOUS` to `LabelName.ABNORMAL`. What do you think?

**Review by samet-akcay — APPROVED** [review]
lgtm, thanks a lot!

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks again
